### PR TITLE
feat: Check if checkov is already installed

### PR DESCRIPTION
**Pull request by gruebel**
- moved the code away from `index.ts`
- minor code improvements

### CODE REVIEW

File: src/report.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export function processReport(report: Report): PolicyValidationPluginReportBeta1
       ruleMetadata: {
         DocumentationUrl: check.guideline,
       },
-    };
+    } satisfies PolicyViolationBeta1;
```
Comment (kadoshms) [inline src/report.ts:67]: nice!!@#@#

File: src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { Report, processReport } from './report';
+import { exec } from './utils';
+
+export interface CheckovValidatorProps {
+  /**
+     * List of checks to run
+     *
+     * @default - all checks are run
+     */
+  readonly check?: string[];
```
Comment (kadoshms) [inline src/plugin.ts:15]: the singular name is a bit confusing IMO - since it's a list I'd expecti to be plular, wdyt?

Comment (gruebel) [reply src/plugin.ts:15]: I know what you mean, I kept it in sync with the naming in the `checkov` CLI. I'll tag @tsmithv11 for his opinion.

Comment (tsmithv11) [reply src/plugin.ts:15]: @kadoshms it's a great callout, but we need to stay consistent with Checkov which has millions of downloads. We can't change the syntax, unfortunately. We may be able to support both in the future, though.

File: src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    this.skipCheck = props.skipCheck ?? [];
+  }
+
+  validate(context: IPolicyValidationContextBeta1): PolicyValidationPluginReportBeta1 {
+    this.templatePaths = context.templatePaths;
+
+    return this.execCheckov();
+  }
+
+  private execCheckov(): PolicyValidationPluginReportBeta1 {
```
Comment (kadoshms) [inline src/plugin.ts]: As we spoke I think a build would work great here as it will ease the interface :)

File: src/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  constructor(props: CheckovValidatorProps = {}) {
+    this.name = 'cdk-validator-checkov';
+
+    this.checkov = 'checkov'; // possible improvement allow Docker usage
+    this.check = props.check ?? [];
+    this.skipCheck = props.skipCheck ?? [];
+  }
+
+  validate(context: IPolicyValidationContextBeta1): PolicyValidationPluginReportBeta1 {
```
Comment (kadoshms) [inline src/plugin.ts:43]: the function name is a bit confusing as in addition to validation it also executes a side effect, just a small suggestion that might work is to remove the `templatePaths` as a class member and use it as a param

File: src/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      const report: Report = exec([this.checkov, ...flags], {
+        json: true,
+        env: {
+          LOG_LEVEL: 'ERROR',
+        },
+      });
+
+      return processReport(report);
+    } catch (e) {
+      console.error(`checkov plugin failed to scan the given templates. Error: ${e}`);
```
Comment (kadoshms) [inline src/plugin.ts]: error might not be printed nicely - worth checking

File: src/plugin.ts (hunk 3 above)
Comment (gruebel) [reply src/plugin.ts]: yeah, that's true. planned for a follow-up PR 🙂

File: src/plugin.ts (hunk 5 above)
Comment (gruebel) [reply src/plugin.ts]: checked, looks good enough 😄

**Pull request merged**